### PR TITLE
Support `onViewStateChange` in InteractiveMap and NavigationControl

### PR DESCRIPTION
**Pull request by ibgreen**
Enables significant alignment across DeckGL examples.

### CODE REVIEW

File: src/mapbox/mapbox.js [hunk 1]
```diff
@@ -169,7 +169,7 @@ export default class Mapbox {
       // TODO - need to call onload again, need to track with Promise?
       props.onLoad();
     } else {
-      const mapOptions = {
+      const mapOptions = Object.assign({}, props, this._getMapboxViewStateProps(props), {
```
Comment (Pessimistress) [inline src/mapbox/mapbox.js]: What's the purpose of this if it's going to be overwritten immediately?

File: src/mapbox/mapbox.js [hunk 2]
```diff
@@ -254,7 +254,9 @@ export default class Mapbox {
 
   // Note: needs to be called after render (e.g. in componentDidUpdate)
   _updateMapSize(oldProps, newProps) {
-    const sizeChanged = oldProps.width !== newProps.width || oldProps.height !== newProps.height;
+    const newSize = this._getSize(newProps);
+    const oldSize = this._getSize(oldProps);
```
Comment (Pessimistress) [inline src/mapbox/mapbox.js]: Why not just use default props?

**Comment by ibgreen** [comment]
Nits addressed

> I don't think we should promote this API in the examples

Makes sense. Example changes reverted

> TransitionManager also needs to be updated.

Good catch. I made the obvious addition.

> Docs

A basic set of doc updates are included. Let me know if more is needed.

File: src/components/interactive-map.js [hunk 3]
```diff
@@ -202,7 +201,7 @@ export default class InteractiveMap extends PureComponent {
   }
 
   componentWillUpdate(nextProps) {
-    this._mapControls.setOptions(nextProps);
+    this._mapControls.setOptions(Object.assign({}, nextProps, nextProps.viewState));
     this._transitionManager.processViewportChange(nextProps);
```
Comment (Pessimistress) [inline src/components/interactive-map.js]: Handle `viewState`

**Pull request merged**